app/models/moment.rb

```diff
@@ -46,15 +55,18 @@ def array_data
     self.strategy = strategy.collect(&:to_i) if strategy.is_a?(Array)
   end
 
-  def category_name
-    category.try(:name)
-  end
+  delegate :category_name, to: :category
```

Comment:
Ooh I like this!

Reply:
Does it work though 🤔 
Admittedly, I rarely use Rails 5 but I thought you need to supply `prefix: true` if the method name would be different on the object.
For instance:
```
delegate :name, to: :category, prefix: true
```

Am I wrong?

Reply:
I'm actually confused by the original code.
Category, mood, and strategy are all arrays of integers; what is calling these methods and how are they in a position to return a name?

Reply:
Oh so this was being used with autocomplete in the Moments and Strategy editors.

`app/views/shared/_autocomplete_checkboxes.html.erb`

```
<%= f.text_field "#{data_type}_name", data: { autocomplete_source: all_data.map(&:name), autocomplete_source_ids: all_data.map(&:id) }, class: 'full_field smaller_margin_top smaller_margin_bottom', placeholder: t('common.form.press_enter') %>
```

Reply:
Does this mean that somehow the array of integers is being converted into an array of objects 😨

Reply:
Yeah... :/ This is my doing lol. I used jquery autocomplete rather than writing my own implementation.